source/uwp/Visualizer/ViewModel/GenericDocumentViewModel.cs

```diff
@@ -117,15 +120,58 @@ protected async Task LoadFromFileAsync(IStorageFile file, string token, bool ass
             }
         }
 
+        private async Task CreateMessageDialog(string text)
+        {
+            var dontWait = new MessageDialog(text);
+            dontWait.CancelCommandIndex = 0;
+            await dontWait.ShowAsync();
+        }
+
+        private async Task SaveNewFile()
+        {
+            var savePicker = new FileSavePicker();
+            savePicker.SuggestedStartLocation = PickerLocationId.DocumentsLibrary;
+            savePicker.FileTypeChoices.Add("Plain Text", new List<string>() { ".json" });
+            savePicker.SuggestedFileName = "NewAdaptiveCard.json";
+
+            File = await savePicker.PickSaveFileAsync();
+            if (File != null)
+            {
+                CachedFileManager.DeferUpdates(File);
+                await FileIO.WriteTextAsync(File, File.Name);
+                FileUpdateStatus status = await CachedFileManager.CompleteUpdatesAsync(File);
+                if (status == FileUpdateStatus.Complete)
+                {
+                    this.Name = File.Name;
+                    await CreateMessageDialog("File " + File.Name + " was saved");
+                }
+                else
+                {
+                    await CreateMessageDialog("File " + File.Name + " couldn't be saved");
+                }
+            }
+            else
+            {
+                await CreateMessageDialog("Save new file was cancelled");
+            }
+        }
+
         public async Task SaveAsync()
         {
-            try
+            if (File == null)
             {
-                await FileIO.WriteTextAsync(File, Payload);
+                await SaveNewFile();
```

Comment:
If this fails (the case when you click on Save without hitting the new file workflow) you'll continue to see the default messagebox with only the Close button and no Esc/cancel handler i think since it's not inside of a catch block right?